### PR TITLE
OBPIH-4136 Corrected list of items being able to add to return

### PR DESCRIPTION
**Pull request by MagdaToczek**

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy [hunk 1]
```diff
@@ -819,23 +819,18 @@ class ProductAvailabilityService {
         Date expirationDate = params.expirationDate ? dateFormat.parse(params.expirationDate) : null
         return ProductAvailability.createCriteria().list {
             eq("location", location)
-            binLocation {
-                ne("locationType", LocationType.get(Constants.RECEIVING_LOCATION_TYPE_ID))
```
Comment (awalkowiak) [inline grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy:823]: Why did you remove this part?

File: grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy [hunk 2]
```diff
@@ -819,23 +819,18 @@ class ProductAvailabilityService {
...
+            if (product) {
+                eq("product", product)
             }
-            or {
```
Comment (awalkowiak) [inline grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy:825]: Why did you remove the `OR`?

File: grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy (hunk 1 above)
Comment (awalkowiak) [reply grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy:823]: Oh, I see that this was requested. Nvm then.

File: grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy (hunk 2 above)
Comment (MagdaToczek) [reply grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy:825]: By mistake.

**Review by pmuchowski — APPROVED** [review]
